src/expr/src/expr/expr_array.rs

```diff
@@ -62,6 +62,15 @@ impl Expression for ArrayExpression {
             .try_for_each(|row| builder.append_row_ref(row))?;
         builder.finish().map(|a| Arc::new(ArrayImpl::List(a)))
     }
+
+    fn eval_row_ref(&self, input: &Row) -> Result<Datum> {
```

Comment:
How about naming it `eval_row`? We already have another struct named `RowRef` that differs from `&Row`. cc @TennyZhuang

Reply:
+1, `RowRef` refers to a row in a DataChunk, which may be misleading here.